pkg/virt-controller/watch/migration.go

```diff
@@ -922,6 +920,7 @@ func (c *MigrationController) handlePendingPodTimeout(migration *virtv1.VirtualM
 	secondsSpentPending := timeSinceCreationSeconds(&pod.ObjectMeta)
 
 	if isPodPendingUnschedulable(pod) {
+		c.triggerAlertForHostModelUnschedulable(vmi, pod)
```

Comment:
I am not sure I understand the reasoning of https://github.com/kubevirt/kubevirt/pull/6918/commits/f5bb8e9a1b5182a03b57946e7c0bc9df921db945 . Why doesn't the execution continue to `updateStatus`?

Reply:
For example, if the timeout exceeded we would delete target pod and have an alert saying VMI cannot be scheduled, then the logic on `updateStatus()` becomes irrelevant. The right behavior here is that even if timeout exceeded and target pod is deleted we want the more precise alert (about host-model) and not the more general alert (about not being able to schedule)